python/tests/test_ops.py

```diff
@@ -1762,8 +1762,8 @@ def test_repeat(self):
         )
 
     def test_tensordot(self):
-        # No fp16 matmuls on linux
-        if self.is_linux:
+        # No fp16 matmuls on cpu
+        if mx.default_device() == mx.cpu:
```

Comment:
I don't think that's right actually, we can do fp16 with BNNS on the CPU 🤔 , we shouldn't stop testing that.

It might be good to check the platform instead.

Reply:
This is where throws:
https://github.com/ml-explore/mlx/blob/28fcd2b519f0fabcd681f2c33e14d71983cad819/mlx/backend/common/default_primitives.cpp#L169-L176

Reply:
Oh I know that on the common backend there is no fp16 but on the accelerate backend for the CPU (which is what is used on Apple silicon devices) we support fp16 and that test should run on the CPU:

https://github.com/ml-explore/mlx/blob/main/mlx/backend/accelerate/matmul.cpp#L201-L206

Reply:
You could set a variable:

```
@property
def is_apple_silicon(self):
    return platform.machine() == "arm64" and platform.system() == "Darwin"
```

Reply:
Ah I forgot the Accelerate backend, I have updated the code.